ssl/quic/quic_impl.c

```diff
@@ -1707,6 +1729,33 @@ static SSL *quic_conn_stream_new(QCTX *ctx, uint64_t flags, int need_lock)
         goto err;
     }
 
+    if (!advance
+        && !ossl_quic_channel_is_new_local_stream_admissible(qc->ch, is_uni)) {
+        struct quic_new_stream_wait_args args;
+
+        /*
+         * Stream count flow control currently doesn't permit this stream to be
+         * opened.
+         */
+        if (no_blocking || !qc_blocking_mode(qc)) {
+            QUIC_RAISE_NON_NORMAL_ERROR(ctx, SSL_R_STREAM_COUNT_LIMITED, NULL);
```

Comment:
This feels like a `SSL_get_error` thing, so maybe it should be considered an IO error.

Reply:
It seems this comment is still unresolved, one way or the other - the errors need updating following the "more errors" PR.

If this is actually a `SSL_get_error()` thing and is an IO issue then this actually raises questions with our API, i.e. `SSL_get_error()` expects you to supply the return code from the function that failed. But here we're talking about `SSL_new_stream()` which returns an `SSL` object. We'd also need to update the documentation accordingly to tell people to call `SSL_get_error()`. Similar thinking for `SSL_accept_stream()`.

Reply:
So this raises an interesting API gap: we have `SSL_get_error` for 'I/O' errors but it seems it can't really be used with functions which return a pointer, even if those are 'I/O' functions. Hmm.

Anyway, I don't mind making this a non-I/O error for now.